src/Bicep.LangServer/Handlers/BicepGetResourceTypesHandler.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Bicep.Types.Az;
+using Bicep.Core.Analyzers.Linter.ApiVersions;
+using Bicep.Core.Configuration;
+using Bicep.Core.Features;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Providers.Az;
+using Bicep.LanguageServer.CompilationManager;
+using MediatR;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    // TODO: Update to receive a textDocument to use current document features to find what is available. 
+    // [Method("textDocument/getResourceCatalog", Direction.ClientToServer)]
+    // public record BicepResourceTypeParams(TextDocumentIdentifier TextDocument) : ITextDocumentIdentifierParams, IRequest<BicepResourceTypes?>;
+    [Method("bicep/getResourceCatalog", Direction.ClientToServer)]
+    public record BicepResourceTypeParams : IRequest<List<BicepResourceGroup>?>;
+
+    public record BicepResourceType(string resourceType, string? apiVersion);
+
+    public record BicepResourceGroup(string group, IEnumerable<BicepResourceType> resourceTypes);
```

Comment:
Can we rename this to `BicepResourceTypeGroup` to avoid any confusion with an Azure resource group?